10up/wp-vulnerability-scanner
# Add Wordfence Intelligence CE as an API provider

### Description of the Change
PR adds support for Wordfence API to check reported vulnerabilities for WordPress core, installed plugins and themes.

Users can choose Wordfence API for vulnerabilities scan by adding the `VULN_API_PROVIDER` constant in wp-config.php

```
define( 'VULN_API_PROVIDER', 'wordfence' );
```

Apart from Wordfence Intelligence CE support, PR makes the below changes.
-  Fixed PHPCS errors
- Added workflow for PHPCS
- Added Build release zip workflow (to attach the zip to release and manually trigger build zip)
- Added Stable build workflow to update the `stable` branch with built code on push to the `trunk` branch.
- Divided tests for each API provider as a separate check

Closes #76 

### How to test the Change
1. Add a `VULN_API_PROVIDER` constant in wp-config.php as below.
```
define( 'VULN_API_PROVIDER', 'wordfence' );
```
2. Try the given `vuln` commands in [readme](https://github.com/10up/wpcli-vulnerability-scanner#usage) to scan for vulnerabilities.
4. Make sure all commands are working fine.
4. Regression: Make sure all commands are working fine with other services (WPScan and Patchstack)

### Changelog Entry

> Added - Wordfence Intelligence CE as an API provider.


### Credits

Props @iamdharmesh @jeffpaul [Charles Sweethill](https://github.com/wordfence), [Matt Barry](https://github.com/wordfence) (Wordfence Team)

### Checklist:
